CanHazFunny/CanHazFunny/JokeService.cs

```diff
@@ -1,15 +1,16 @@
 ﻿using System.Net.Http;
+using System.Text.Json;
 
 namespace CanHazFunny
 {
-    public class JokeService
+    public class JokeService : IJokeService
     {
         private HttpClient HttpClient { get; } = new();
 
         public string GetJoke()
         {
-            string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
-            return joke;
+            JokeJson jokejson = JsonSerializer.Deserialize<JokeJson>(HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api?format=json").Result);
```

Comment:
This works, but there is a convenience method that you can use too. You can see an example is the [solution I posted](https://github.com/IntelliTect-Samples/EWU-CSCD371-2021-Winter/blob/Assignment3Complete/CanHazFunny/CanHazFunny/JokeService.cs#L12)